lib/porky_lib/file_service.rb

```
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'aws-sdk-s3'
+require 'singleton'
+
+class PorkyLib::FileService
+  include Singleton
+
+  def read(bucket_name, file_key, options = {})
+    tempfile = Tempfile.new
+    s3.bucket(bucket_name).object(file_key).download_file(tempfile.path, options)
+
+    decrypt_file_contents(tempfile)
+  end
+
+  def write(file, bucket_name, key_id, options = {})
+    return if file.nil? || bucket_name.nil? || key_id.nil?
```

Do we want to throw an exception here?